pkg/client/cache/listers.go

```diff
@@ -670,3 +671,38 @@ func (s *StoreToPetSetLister) GetPodPetSets(pod *api.Pod) (psList []apps.PetSet,
 	}
 	return
 }
+
+type StoreToPodDisruptionBudgetLister struct {
+	Store
+}
+
+func (s *StoreToPodDisruptionBudgetLister) GetPodPodDisruptionBudgets(pod *api.Pod) (pdbList []policy.PodDisruptionBudget, err error) {
+	var selector labels.Selector
+	var pdb policy.PodDisruptionBudget
+
+	if len(pod.Labels) == 0 {
+		err = fmt.Errorf("no PodDisruptionBudgets found for pod %v because it has no labels", pod.Name)
+		return
+	}
+
+	for _, m := range s.Store.List() {
+		pdb = *m.(*policy.PodDisruptionBudget)
+		if pdb.Namespace != pod.Namespace {
+			continue
+		}
+		selector, err = unversioned.LabelSelectorAsSelector(pdb.Spec.Selector)
+		if err != nil {
+			err = fmt.Errorf("invalid selector: %v", err)
+			return
+		}
+
+		if selector.Empty() || !selector.Matches(labels.Set(pod.Labels)) {
```

Comment:
I'll add a comment, but in this case I'm copying behavior from other controllers.
https://github.com/kubernetes/kubernetes/blob/master/pkg/client/cache/listers.go#L254

Search for "nil or empty" in that file.

IIRC, during validation I verify that we have a non-empty selector, so this is sort of defense-in-depth.